### PR TITLE
minor changes for compatibility with newer numpy versions

### PR DESCRIPTION
**Pull request by mskarl**
Includes few changes in types for newer numpy versions.

### CODE REVIEW

File: skyllh/core/analysis.py [hunk 1]
```diff
@@ -1034,7 +1034,7 @@ def do_trials(
 
         result_dtype = result_list[0].dtype
         result = np.empty(n, dtype=result_dtype)
-        result[:] = result_list[:]
+        result[:] = result_list[:][0]
```
Comment (martwo) [inline skyllh/core/analysis.py:1037]: Can you please explain why this change is required?

File: skyllh/analyses/i3/publicdata_ps/signal_generator.py [hunk 2]
```diff
@@ -85,6 +85,8 @@ def _generate_inv_cdf_spline(self, flux_model, log_e_min,
         prob_per_bin /= np.sum(prob_per_bin)
 
         # Compute the cumulative distribution CDF.
+    #    print(prob_per_bin)
```
Comment (martwo) [inline skyllh/analyses/i3/publicdata_ps/signal_generator.py:88]: These comment lines should not be added.

**Review by martwo — COMMENTED** [review]
I don't understand the change in the do_trials method. Why is that?

**Comment by mskarl** [comment]
> 
threw an error without it. Only for nympy 1.24.

**Comment by mskarl** [comment]
> > 
> 
> threw an error without it. Only for nympy 1.24.
with the newest nersion I suddenly got this: 

result[:] = result_list[:]
ValueError: setting an array element with a sequence. The requested array would exceed the maximum number of dimension of 1.

it did work with older versions.

**Comment by martwo** [comment]
I investigated this and it seem one has to create an array of the result list first. So the proper line would be:
`result[:] = np.array(result_list)[:, 0]`

The current fix would broadcast the first result across all other results!

**Comment by mskarl** [comment]
> I investigated this and it seem one has to create an array of the result list first. So the proper line would be: `result[:] = np.array(result_list)[:, 0]`
> 
> The current fix would broadcast the first result across all other results!

Right, I changed it.

**Review by martwo — APPROVED** [review]
